### PR TITLE
Migrate to Ballerina Swan Lake

### PR DESCRIPTION
**Pull request by Maninda**
## Purpose
Migrate the connector to new Ballerina version, "Swan Lake". Fixes https://github.com/ballerina-platform/module-ballerinax-aws.sqs/issues/15 .

### CODE REVIEW

File: src/aws.sqs/amazonsqs_errors.bal [hunk 1]
```diff
@@ -25,54 +25,31 @@ public type ErrorDetail record {
...
 
-public const ERROR_CLIENT = "{ballerinax/aws.sqs}ErrorClient";
-public type ErrorClient error<ERROR_CLIENT, ErrorDetail>;
-
-public const ERROR_SERVER = "{ballerinax/aws.sqs}ErrorServer";
-public type ErrorServer error<ERROR_SERVER, ErrorDetail>;
-
-public const ERROR_OPERATION = "{ballerinax/aws.sqs}ErrorOperation";
-public type ErrorOperation error<ERROR_OPERATION, ErrorDetail>;
+public type ErrorOperation distinct error;
```
Comment (chamil321) [inline src/aws.sqs/amazonsqs_errors.bal:34]: Mostly when naming error, the `Error` has been used as a suffix along with the error context. Shall we change this name as follows?
```suggestion
public type OperationError distinct error;
```

File: src/aws.sqs/amazonsqs_errors.bal [hunk 2]
```diff
@@ -25,54 +25,31 @@ public type ErrorDetail record {
...
 
 // AWS SQS Data Mappings Error Types
 
-public const ERROR_DATA_MAPPING = "{ballerinax/aws.sqs}ErrorDataMapping";
-public type ErrorDataMapping error<ERROR_DATA_MAPPING, ErrorDetail>;
+public type ErrorDataMapping distinct error;
```
Comment (chamil321) [inline src/aws.sqs/amazonsqs_errors.bal:38]:
```suggestion
public type DataMappingError distinct error;
```

File: src/aws.sqs/amazonsqs_errors.bal [hunk 3]
```diff
@@ -25,54 +25,31 @@ public type ErrorDetail record {
...
 
 public const CONVERT_XML_TO_OUTBOUND_MESSAGE_FAILED = "{ballerinax/aws.sqs}ConvertXmlToOutboundMessageFailed";
```
Comment (chamil321) [inline src/aws.sqs/amazonsqs_errors.bal:40]: Do we need these error reasons anymore?

File: .travis.yml [hunk 4]
```diff
@@ -13,8 +13,8 @@
 # limitations under the License.
 
 script:
-  - wget https://product-dist.ballerina.io/downloads/1.2.0/ballerina-linux-installer-x64-1.2.0.deb
-  - sudo dpkg -i ballerina-linux-installer-x64-1.2.0.deb
+  - wget https://product-dist.ballerina.io/downloads/2.0.0/ballerina-linux-installer-x64-2.0.0.deb
+  - sudo dpkg -i ballerina-linux-installer-x64-2.0.0.deb
```
Comment (daneshk) [inline .travis.yml:17]: Please change this to the latest link

File: Ballerina.lock [hunk 5]
```diff
@@ -1,4 +1,344 @@
 org_name = "ballerinax"
-version = "0.11.0"
+version = "0.12.0"
```
Comment (daneshk) [inline Ballerina.lock:2]: the version number should be 0.99.0 for the preview release

File: Ballerina.toml [hunk 6]
```diff
@@ -2,6 +2,6 @@
 org-name= "ballerinax"
 license= ["Apache-2.0"]
 authors = ["Ballerina"]
-version= "0.11.0"
-keywords = ["amazonsqs", "Amazon", "SQS", "Ballerina", "connector", "client"]
-repository = "https://github.com/ballerina-platform/module-amazonsqs"
+version= "0.12.0"
```
Comment (daneshk) [inline Ballerina.toml:5]: the version number should be 0.99.0 for the preview release

File: README.md [hunk 7]
```diff
@@ -7,7 +7,7 @@ Amazon SQS Connector allows you to connect to the Amazon Simple Queue Service (S
 ## Compatibility
 | Ballerina Language Versions | Amazon SQS API version  |
 | --------------------------- | ----------------------  |
-|            1.2.x            | 2012-11-05              |
+|            2.0.0            | 2012-11-05              |
```
Comment (daneshk) [inline README.md:10]: Please change it to the correct one. `Swan Lake Preview1`

File: src/aws.sqs/Module.md [hunk 8]
```diff
@@ -7,7 +7,7 @@ AWS SQS Connector allows you to connect to the Amazon SQS service via REST API f
 
 |                    |    Versions     |  
 |:------------------:|:---------------:|
-| Ballerina Language |   1.2.x         |
+| Ballerina Language |   2.0.0         |
```
Comment (daneshk) [inline src/aws.sqs/Module.md:10]: Please change it to the correct one. `Swan Lake Preview1`

File: src/aws.sqs/amazonsqs_errors.bal (hunk 1 above)
Comment (Maninda) [reply src/aws.sqs/amazonsqs_errors.bal]: Changed in all the instances.

File: src/aws.sqs/amazonsqs_errors.bal (hunk 2 above)
Comment (Maninda) [reply src/aws.sqs/amazonsqs_errors.bal]: Changed in all the instances.

File: src/aws.sqs/amazonsqs_errors.bal (hunk 3 above)
Comment (Maninda) [reply src/aws.sqs/amazonsqs_errors.bal]: Removed all the unnecessary error reasons.

File: Ballerina.lock (hunk 5 above)
Comment (Maninda) [reply Ballerina.lock]: Changed.

File: Ballerina.toml (hunk 6 above)
Comment (Maninda) [reply Ballerina.toml]: Changed.

File: README.md (hunk 7 above)
Comment (Maninda) [reply README.md]: Changed.

File: src/aws.sqs/Module.md (hunk 8 above)
Comment (Maninda) [reply src/aws.sqs/Module.md]: Changed.

File: .travis.yml (hunk 4 above)
Comment (Maninda) [reply .travis.yml]: Corrected.

**Pull request merged**